core/src/main/java/org/axonframework/deadline/SimpleDeadlineManager.java

```diff
@@ -161,31 +171,40 @@ private void cancelSchedule(DeadlineId deadlineId) {
 
         private final String deadlineName;
         private final ScopeDescriptor deadlineScope;
-        private final Object messageOrPayload;
+        private final DeadlineMessage<?> deadlineMessage;
         private final String deadlineId;
 
         private DeadlineTask(String deadlineName,
                              ScopeDescriptor deadlineScope,
-                             Object messageOrPayload,
+                             DeadlineMessage<?> deadlineMessage,
                              String deadlineId) {
             this.deadlineName = deadlineName;
             this.deadlineScope = deadlineScope;
-            this.messageOrPayload = messageOrPayload;
+            this.deadlineMessage = deadlineMessage;
             this.deadlineId = deadlineId;
         }
 
         @Override
         public void run() {
-            DeadlineMessage<?> deadlineMessage =
-                    GenericDeadlineMessage.asDeadlineMessage(deadlineName, messageOrPayload);
             if (logger.isDebugEnabled()) {
                 logger.debug("Triggered deadline");
             }
 
             try {
                 UnitOfWork<DeadlineMessage<?>> unitOfWork = new DefaultUnitOfWork<>(deadlineMessage);
                 unitOfWork.attachTransaction(transactionManager);
-                unitOfWork.execute(() -> executeScheduledDeadline(deadlineMessage, deadlineScope));
+                InterceptorChain chain =
+                        new DefaultInterceptorChain<>(unitOfWork,
+                                                      handlerInterceptors(),
+                                                      dm -> {
```

Comment:
When I read `dm` I can only think of 'Dungeon Master'..personally I prefer to write out parameters for lambda function.